src/components/createContainer.jsx

```diff
@@ -74,29 +82,42 @@ export default function createContainer(DecoratedComponent, specs) {
       this.unsubscribe();
     }
 
+    shouldComponentUpdate(nextProps, nextState){
```

Comment:
In ```shouldComponentUpdate()```

* If upstream props change then we will update (eventually trigging arg updates).
* The adaptor API is given an opportunity to determine if args/data have actually changed.
```
// A much stronger, deeper comparison might be necessary if data props or args are complex objects (or arrays)
adaptor.shouldComponentUpdate = _.isEqual
```